grpcutil/status.go

```diff
@@ -0,0 +1,60 @@
+package grpcutil
+
+import (
+	"context"
+	"errors"
+
+	"github.com/gogo/status"
+	"google.golang.org/grpc/codes"
+	grpcstatus "google.golang.org/grpc/status"
+)
+
+// ErrorToStatus tries to cast the given error into gogo's status.Status.
```

Comment:
This description seems wrong: there's no casting to gogo's status.Status. Code only checks for `GRPCStatus()` method with specific return type.

I'd suggest to remove this comment, or to update it:

```golang
// ErrorToStatus returns a *github.com/gogo/status.Status representation of err.
//
//   - If err implements the method `GRPCStatus() *google.golang.org/grpc/status.Status` and
//     `GRPCStatus()` does not return nil, or if err wraps a type satisfying this, Status from `GRPCStatus()` is
//     converted to gogo Status, and returned. In that case, ok is true.
//
//   - If err is or GRPCStatus() returns nil, a nil Status is returned and ok is false.
//
//   - Otherwise, err is an error not compatible with this function. In this
//     case, a nil Status is returned and ok is false.
```